R2API.Stages/StageRegistration.cs

```diff
@@ -173,18 +189,20 @@ public static void PrintSceneVariants(string key)
         }
 
         StageRegistration.SetHooks();
-        foreach(SceneDef sceneDef in stageVariantDictionary[key])
+        foreach (SceneDef sceneDef in stageVariantDictionary[key])
         {
             StagesPlugin.Logger.LogDebug($"{sceneDef.cachedName}, baseSceneName: {sceneDef.baseSceneName}");
         }
     }
 
     /// <summary>
     /// Registers the SceneDef into the loop. Any SceneDef registered with the same baseSceneName as another SceneDef will be counted as a variant.
-    /// Variants will have their weights split equally amongst each other. Don't use this method for scenes that aren't part of the loop (Stages 1-5).
     /// </summary>
     /// <param name="sceneDef">The SceneDef being registered</param>
-    public static void RegisterSceneDefToLoop(SceneDef sceneDef)
+    /// <param name="weight">The weight of the SceneDef being rolled in the SceneCollection</param>
+    /// <param name="preLoop">If the stage will roll pre-loop: 1 <= current stage <= 5 </param>
+    /// <param name="postLoop">If the stage will roll post-loop: current stage > 5 </param>
+    public static void RegisterSceneDefToLoop(SceneDef sceneDef, float weight = defaultWeight, bool preLoop = true, bool postLoop = true)
```

Comment:
this is a binary level breaking change afaik, people will need to recompile against the new dll for their mod to work again

Reply:
That explains a lot because I had that issue, I didn't know it was from that. What do you suggest I do instead?

Reply:
keep the old one, and add to it the Obsolete attribute to it with a message to use the new one with the extra parameters